.github/workflows/preview-image.yml

```diff
@@ -71,7 +69,12 @@ jobs:
         run: |
           set -x
           VERSION=$(jq -r .version package.json)
-          FULL_VERSION=${VERSION}-b${GITHUB_RUN_ID}.${GITHUB_RUN_NUMBER}
+          BRANCH=${{ event.workflow_run.head_branch }}
+          if [[ "$BRANCH" =~ ^[0-9]+\.[0-9]+\.[0-9]+\-dev ]]; then
+            FULL_VERSION=${VERSION}
+          else
+            FULL_VERSION=${VERSION}-b${GITHUB_RUN_ID}.${GITHUB_RUN_NUMBER}
+          fi
```

Comment:
@eradman Reading over these lines here, doesn't the `FULL_VERSION` variable end up how it used to be for the monthly snapshots?

Reply:
Previously the trigger condition was based on a single condition

```yml
name: Preview Image
on:
  push:
    tags:
      - '*-dev'
```

This made the entire workflow easy to reason about. I don't know what will happen when the workflow is run; needs to be tested in a private repository.

Reply:
it's needed to:
1.  prevent building image with code, that doesn't pass tests in master
2. spend less time on ARM64 images build, which are only possible to build on QEMU in github actions

Reply:
it doesn't solve a problem, still emulation is required